File: LMS/LMS.UI/huda-center/src/app/shared/models/hr/leaves/leave-details.dto.ts
import { LeaveStatus } from "./leave-status.enum";
import { LeaveType } from "./leave-type.enum";

export interface LeaveDetailsDto {
  leaveId: string;
  employeeId: string;
  employeeName: string;
  startDate: Date;
  endDate: Date;
  leaveType: LeaveType;
  leaveStatus: LeaveStatus;
  isPaid: boolean;
  reason: string;
  createdAt: Date; 
  updatedAt: Date;
}

File: LMS/LMS.Application/DTOs/HR/LeaveDetailsDto.cs
using LMS.Domain.HR.Enums;

namespace LMS.Application.DTOs.HR
{
    public class LeaveDetailsDto
    {
        public Guid LeaveId { get; set; }
        public Guid EmployeeId { get; set; }
        public string EmployeeName { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public LeaveType LeaveType { get; set; }
        public LeaveStatus LeaveStatus { get; set; }
        public bool IsPaid { get; set; }
        public string Reason { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}